src/ProtocolStreamReader.php

```diff
@@ -30,33 +30,32 @@ public function __construct($input)
     {
         $this->input = $input;
         Loop\addReadStream($this->input, function() {
-            while (($c = fgetc($this->input)) !== false) {
-                $this->buffer .= $c;
-                switch ($this->parsingMode) {
-                    case ParsingMode::HEADERS:
-                        if ($this->buffer === "\r\n") {
-                            $this->parsingMode = ParsingMode::BODY;
-                            $this->contentLength = (int)$this->headers['Content-Length'];
-                            $this->buffer = '';
-                        } else if (substr($this->buffer, -2) === "\r\n") {
-                            $parts = explode(':', $this->buffer);
-                            $this->headers[$parts[0]] = trim($parts[1]);
-                            $this->buffer = '';
+            $c = fgetc($this->input);
```

Comment:
Could you commend a bit on this change?
As I understand [`Loop\addReadStream`](http://sabre.io/event/loop/), it calls the callback as soon as the stream has data in the buffer using `stream_select` under the hood. So as long as there is no data, it will not block. When the callback is called, it could be that there is more than one character in the buffer, so I call `fgetc` (blocking) until the buffer is empty again. I have to admit I was/am not familiar at all with what PHP stream read functions are suitable here as I never worked with them. I used `fgetc` because I wanted to be 100% sure I don't loose any data and the other functions seemed to only read a chunk of a specific size (while here `Loop\addReadStream` should already take care of the chunking)